pkg/planner/core/find_best_task.go

```diff
@@ -2617,13 +2617,13 @@ func convertToTableScan(ds *logicalop.DataSource, prop *property.PhysicalPropert
 			ts.AnnIndexExtra = &VectorIndexExtra{
 				IndexInfo: candidate.path.Index,
 				PushDownQueryInfo: &tipb.ANNQueryInfo{
-					QueryType:      tipb.ANNQueryType_OrderBy,
-					DistanceMetric: tipb.VectorDistanceMetric(distanceMetricPB),
-					TopK:           prop.VectorProp.TopK,
-					ColumnName:     ts.Table.Columns[candidate.path.Index.Columns[0].Offset].Name.L,
-					ColumnId:       prop.VectorProp.Column.ID,
-					IndexId:        candidate.path.Index.ID,
-					RefVecF32:      prop.VectorProp.Vec.SerializeTo(nil),
+					QueryType:          tipb.ANNQueryType_OrderBy,
+					DistanceMetric:     tipb.VectorDistanceMetric(distanceMetricPB),
+					TopK:               prop.VectorProp.TopK,
+					ColumnName:         ts.Table.Columns[candidate.path.Index.Columns[0].Offset].Name.L,
+					DeprecatedColumnId: &prop.VectorProp.Column.ID,
+					IndexId:            candidate.path.Index.ID,
+					RefVecF32:          prop.VectorProp.Vec.SerializeTo(nil),
```

Comment:
```suggestion
					RefVecF32:          prop.VectorProp.Vec.SerializeTo(nil),
					Column: *tidbutil.ColumnToProto(prop.VectorProp.Column.ToInfo(), false, false),
```
`Column` can also be set here for compatibility